core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSourceProvider.java

```diff
@@ -17,6 +17,14 @@
 
 public interface ConnectorPageSourceProvider
```

Comment:
That's possible but it will require updating all the connector implementations. Currently, we rely on the default implementation for all but the Iceberg connector so this would require a large refactor commit.  
Do you prefer we do it that way? 

Another option is for `PageSourceManager` to check if the implementation is a factory or not at runtime.

Reply:
Seems resonable.  The simple backwards solution is to add a new interface and then modify `Connector` with a fallback.  Like this:
```java
public interface ConnectorPageSourceProviderFactory
        extends Closeable
{
    ConnectorPageSourceProvider createPageSourceProvider();

    @Override
    default void close() {}
}

    /**
     * @throws UnsupportedOperationException if this connector does not support reading tables page at a time
     */
    default ConnectorPageSourceProviderFactory getPageSourceProviderFactory()
    {
        ConnectorPageSourceProvider pageSourceProvider = getPageSourceProvider();
        return () -> pageSourceProvider;
    }
```

Reply:
Alright, I went ahead and added this to the `Connector` class along with the new Interface.